### PR TITLE
[LTS] Use public client auth for odsp e2e flows (#21091)

### PR DESCRIPTION
**Pull request by Abe27342**
## Description

Currently, we use the ropc flow with a confidential client to authenticate in our e2e/stress tests against odsp. This is less appropriate than using a public client flow, since our application registration really only needs to have delegated permissions.
This PR adjusts things to use a public flow--I've updated the relevant app registrations to allow both forms of auth already.

Using a public flow also means our infrastructure setup here aligns exactly with [this relatively recent MSFT guidance](https://learn.microsoft.com/en-us/entra/identity-platform/test-automate-integration-testing?tabs=dotnet).

Cherry-pick of #21091. This change needs to be in every branch we plan on maintaining real service tests against odsp for, as I plan on removing the application configuration enabling confidential client auth.

### CODE REVIEW

**Comment by Abe27342** [comment]
needed to make a bunch of manual updates here b/c of differences in formatting etc.

File: packages/utils/odsp-doclib-utils/src/odspAuth.ts [hunk 1]
```diff
@@ -13,6 +13,13 @@ export interface IOdspTokens {
     readonly refreshToken: string;
 }
 
+/**
+ * Configuration for a public client.
+ */
+export interface IPublicClientConfig {
```
Comment (Abe27342) [inline packages/utils/odsp-doclib-utils/src/odspAuth.ts:19]: typetests don't support removing interfaces on LTS.

**Review by anthony-murphy — APPROVED** [review]

File: packages/utils/odsp-doclib-utils/src/odspAuth.ts (hunk 1 above)
Comment (markfields) [reply packages/utils/odsp-doclib-utils/src/odspAuth.ts:19]: Deprecate it? Not that it matters

**Review by markfields — APPROVED** [review]
Diff looks fine.  You could also do global search for interesting keywords like `login__microsoft__secret` to make sure there aren't other hits to deal with

**Comment by Abe27342** [comment]
> Diff looks fine. You could also do global search for interesting keywords like `login__microsoft__secret` to make sure there aren't other hits to deal with

yup. I did that. there was in fact a file hanging around which I removed a ref from. (though I think it was one we deleted since it was long unused so probably nbd if I didn't find the reference)

File: packages/utils/odsp-doclib-utils/src/odspAuth.ts (hunk 1 above)
Comment (Abe27342) [reply packages/utils/odsp-doclib-utils/src/odspAuth.ts:19]: yeahhh this is all internal in modern code anyway, so probably we did that at some point.

**Pull request merged**